### PR TITLE
Evagelos/tests refactoring

### PR DESCRIPTION
**Pull request by ev-agelos**

### CODE REVIEW

File: spynl/tests/conftest.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
                                        encode_date)
 
 
-@pytest.fixture(scope="session")
```
Comment (nhoening) [inline spynl/tests/conftest.py]: It seems to me that this would slow down tests which do not parallelize?

File: spynl/tests/conftest.py [hunk 2]
```diff
@@ -55,6 +55,17 @@ def app(settings):
 
 
 @pytest.fixture
+def app_factory():
```
Comment (nhoening) [inline spynl/tests/conftest.py:58]: I don't think we need this. The app_factory I used was simply the conftest.app fixture and I passed in modified settings.

**Review by nhoening — CHANGES_REQUESTED** [review]

File: spynl/tests/conftest.py (hunk 1 above)
Comment (koraa) [reply spynl/tests/conftest.py]: @ev-agelos Did you check the impact on performance? IIRC app creation (down below) takes a lot of time, but I could  be wrong.

File: spynl/tests/test_endpoint_availability.py [hunk 3]
```diff
@@ -39,6 +22,7 @@ def non_production_endpoint(ctx, request):
 
     monkeypatch.setattr(endpoints, 'main', patched_main)
 
+    settings['spynl.spynl_environment'] = 'production'
```
Comment (koraa) [inline spynl/tests/test_endpoint_availability.py]: This is out of scope of this, but IMO if we set the environment to production in non-production settings, we should seriously consider to banish any if statements asking for why env we are in.

**Review by koraa — COMMENTED** [review]
This really is supposed to have a better commit message. Do you mind explaining to me what you refactored? I could come up with a better commit message

Refers to commit "refactoring"

**Review by koraa — COMMENTED** [review]
I have no clue what this does. Please explain and please explain why.

Refers to commit "￼use temp files outside the project for tests"

**Review by koraa — COMMENTED** [review]
This might be a good opportunity to see if we like the way github displays squashes :)
I believe it is much more useful in that regard than bitbucket!

Refers to commit "fixed rebase conflicts, commits will be squashed anyway
"

File: spynl/tests/conftest.py (hunk 1 above)
Comment (ev-agelos) [reply spynl/tests/conftest.py]: Yes guys I need to fix this.

File: spynl/tests/conftest.py (hunk 2 above)
Comment (ev-agelos) [reply spynl/tests/conftest.py:58]: Thing is, i removed __init__py from tests so now you can't import. I did that as PyTest recommends.
I also thought about using imports that might be bad idea when you want to be sure that each test is using it's own fixtures etc, you do not need if for example a test patched the `app` fixture where you import it in 2 or more files and you run the tests in parallel.

**Pull request merged**